spec/test-utils.js

```diff
@@ -1,10 +1,25 @@
 "use strict";
 import expect from 'expect';
+import q from 'q';
 
 const sdk = require("..");
 const MatrixEvent = sdk.MatrixEvent;
 
 /**
+ * Return a promise that is resolved when the client next emits a
+ * SYNCING event.
+ * @param {Object} client The client
+ * @return {Promise} Resolves once the client has emitted a SYNCING event
+ */
+module.exports.syncPromise = function(client) {
+    const def = q.defer();
+    client.on('sync', (state) => {
```

Comment:
Nit: We should probably be using [once](https://nodejs.org/api/events.html#events_emitter_once_eventname_listener) to clean up that listener after it is invoked.